tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,372 @@
+    'vmi_headless_test',
+    'access_test',
+    'storage_test',
+    'container_disk_test',
+    'datavolume_test',
+    'hotplug_test',
+    'vmi_cloudinit_test',
+    'vmi_iothreads_test',
+    'vmi_multiqueue_test',
+    'primary_pod_network_test',
+    'vmi_multus_test',
+    'vmi_networking_test',
+    'networkpolicy_test',
+    'vmi_monitoring_test',
+    'vnc_test',
+    'console_test',
+    'credentials_test',
+    'expose_test',
+    'replicaset_test',
+    'network_services_test',
+    'vmi_kernel_boot_test',
+    'infra_test',
+    'operator_test',
+    'migration_test',
+    'subresource_api_test',
+    'virt_control_plane_test'
+);
+
+sub run {
+    my ($self) = shift;
+    my $sut_ip = get_required_var('SUT_IP');
+
+    set_var('SERVER_IP', $sut_ip);
+    bmwqemu::save_vars();
+
+    # Synchronize the server & agent nodes
+    # mutex_wait('KUBEVIRT_TEST_SETUP');
+    barrier_create('KUBEVIRT_TEST_SETUP', 2);
+    barrier_wait('KUBEVIRT_TEST_SETUP');
+
+    my $agent_ip = $self->get_var_from_parent("AGENT_IP");
+    record_info('Agent IP', $agent_ip);
+
+    record_info('Start RKE2 server setup', '');
+    $self->rke2_server_setup($agent_ip);
+
+    record_info('Deploy kubevirt manifests', '');
+    $self->deploy_kubevirt_manifests();
+
+    record_info('Run kubevirt tests', '');
+    $self->run_virt_tests();
+
+    record_info('Generate test report', '');
+    $self->generate_test_report();
+
+    record_info('Upload results', '');
+    $self->upload_results();
+
+    barrier_create('KUBEVIRT_TEST_DONE', 2);
+    barrier_wait('KUBEVIRT_TEST_DONE');
+}
+
+sub rke2_server_setup {
+    my ($self, $agent_ip) = @_;
+
+    systemctl('stop apparmor.service');
+    systemctl('stop firewalld.service');
+    $self->setup_passwordless_ssh_login($agent_ip);
+
+    # Check if the package 'ca-certificates-suse' are installed on the node
+    ensure_ca_certificates_suse_installed();
+
+    # Install downstream kubevirt packages
+    zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+
+    # RKE2 deployment on server node
+    # Default is to setup service with the latest RKE2 version, the parameter INSTALL_RKE2_VERSION allows to setup with a specified version.
+    my $rke2_ver = get_var('INSTALL_RKE2_VERSION');
+    if ($rke2_ver) {
+        record_info('RKE2 version', $rke2_ver);
+        assert_script_run("curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=$rke2_ver sh -", timeout => 180);
+    } else {
+        assert_script_run('curl -sfL https://get.rke2.io | sh -', timeout => 180);
+    }
+
+    # For network multus backend testing
+    assert_script_run("sed -i '/ExecStart=/s/server\$/server --cni=multus,canal/' /etc/systemd/system/rke2-server.service");
+
+    # Enable rke2-server service
+    systemctl('restart yast-timesync.service');
+    systemctl('enable rke2-server.service');
+    systemctl('start rke2-server.service', timeout => 180);
+    check_service_status();
```

Comment:
I think the way you calling member function is not very consistent. Sometimes $self->xxxx, sometimes just check_service_status.

Reply:
Updated, thanks.